### PR TITLE
pato/mswin conan

### PR DESCRIPTION
**Pull request by aroidzap**
- Conan modules (`BUILDSYS_CONAN` switch)
- Windows build bugfixes
- Windows build is based on Conan modules now

### CODE REVIEW

File: Modules.Conan/FindBoost.cmake [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  set(Boost_IOSTREAMS_FOUND TRUE)
+  set(Boost_IOSTREAMS_LIBRARIES Boost::iostreams)
+  list(APPEND Boost_LIBRARIES Boost::iostreams)
+endif()
+
+if (TARGET Boost::unit_test_framework)
+  set(Boost_UNIT_TEST_FRAMEWORK_FOUND TRUE)
+  set(Boost_UNIT_TEST_FRAMEWORK_LIBRARIES Boost::unit_test_framework)
+  list(APPEND Boost_LIBRARIES Boost::unit_test_framework)
+endif()
```
Comment (zap150) [inline Modules.Conan/FindBoost.cmake:72]: I suggest adding newlines at eof everywhere.

File: Modules.Conan/FindCatch.cmake [hunk 2]
```diff
@@ -0,0 +1,12 @@
+find_package(Catch2 REQUIRED CONFIG)
```
Comment (zap150) [inline Modules.Conan/FindCatch.cmake:1]: Do we use Catch or Catch2?

File: Modules.Conan/FindOpenMesh.cmake [hunk 3]
```diff
@@ -0,0 +1,9 @@
+find_package(OpenMesh 8.1 REQUIRED CONFIG)
+# TODO: Find other solution
+message(WARNING "Force-using OpenMesh 8.1")
```
Comment (zap150) [inline Modules.Conan/FindOpenMesh.cmake:3]: Why?

File: Modules.Conan/FindPDAL.cmake [hunk 4]
```diff
@@ -0,0 +1,9 @@
+find_package(PDAL 2.3 REQUIRED CONFIG)
+# TODO: Find other solution
+message(WARNING "Force-using PDAL 2.3")
```
Comment (zap150) [inline Modules.Conan/FindPDAL.cmake:3]: Which PDAL do we use here?

File: buildsys.cmake [hunk 5]
```diff
@@ -111,7 +111,7 @@ macro(enable_c11)
     set(CMAKE_C_STANDARD 11)
     cpp_msvc_overrides()
   else()
-    message(FATAL_ERROR "Unknown C compiler: CMAKE_C_COMPILER_ID.")
+    message(FATAL_ERROR "Unknown C compiler: ${CMAKE_C_COMPILER_ID}.")
```
Comment (zap150) [inline buildsys.cmake:114]: :D

**Review by zap150 — COMMENTED** [review]
Shouldn't we enforce minimal versions of the libraries or sth like that?

File: Modules.Conan/FindCatch.cmake (hunk 2 above)
Comment (aroidzap) [reply Modules.Conan/FindCatch.cmake:1]: @malytomas used Catch2 for Windows build, since Catch is not available in Conan.

Comment (pato-melowntech) [reply Modules.Conan/FindCatch.cmake]: FindCatch.cmake changed to FindCatch2.cmake

**Review by zap150 — APPROVED** [review]

**Pull request merged**